remedy segfault with openmpi/intelmpi in MPI_Reduc

airmler opened the pull request.

airmler committed:
remedy segfault with openmpi/intelmpi in MPI_Reduc

airmler committed:
1.) contraction time estimate returns value in sec
2.) dgemm take heed of 'rectangularism' of contraction
3.) used realistic model values for vsc4

airmler committed:
comment out one print statement

airmler committed:
undo changes related to time estimates

airmler committed:
add dryRun feature. without any output

airmler committed:
change output for dryRun

airmler committed:
change of model + very lengthy output in dryRun

airmler committed:
minor changes on dryRun output

airmler committed:
solomonik fix: disable fast slice when tensors are not mapped to all processors

solomonik commented:
I am a bit confused about the scope of changes in this request. Seems like mostly stuff for dry-runs, but also changes to model parameters are included (possibly due to more recent CTF commits?). Some tweaks to the time estimates seem like they would affect performance nontrivially, on the other hand if they are corrections/improvements, this might be valuable.

Sorry for not getting to this for such a long time.

airmler commented:
Sorry I messed something up with my pull request. Please dont accept it

airmler committed:
simple test

airmler committed:
blocked communicator for summa2d and matrices only

airmler committed:
profile bcast! temp. as it needs an extra barrier

airmler committed:
fix typo in last commit

airmler committed:
aG:Add --without-scalapack to the configure script

airmler committed:
Merge branch 'master' of https://github.com/airmler/ctf

airmler committed:
add node-awareness. still segfaulting in dryRun

airmler committed:
fix segfaults in dryRun

airmler committed:
add node awareness to dryRun branch

airmler committed:
Merge branch 'master' of https://github.com/airmler/ctf

airmler committed:
Add cc4s-model-trainer and change dump of TUNE data:
- minor changes in the output of dry-run
- switch on exhaustive in (almost) every contraction
- dont concider 'shape' in gemm contractions
- new feature: only add model parameters which were actually tuned
- change: dump only the slowest measurement from all ranks

airmler committed:
Merge branch 'master' of github.com:airmler/ctf

airmler committed:
minor changes in cc4s_model_trainer

airmler committed:
Merge branch 'master' of https://github.com/airmler/ctf
